.expeditor/scripts/end_to_end/run_e2e_test_core.ps1

```diff
@@ -0,0 +1,126 @@
+# This script expects `setup_environment.sh` or `setup_environment.ps1` to be sourced before execution
+
+param (
+    [string]$Channel = "dev",
+    [string]$TestName,
+    [string]$BuilderUrl = $env:HAB_BLDR_URL
+)
+
+$ErrorActionPreference = "stop"
+
+if (!$IsCoreCLR) {
+    Write-Error "End-to-end tests are only compatiable with Powershell Core"
+}
+
+###################################################################################################
+
+$EndToEndTestingOrigin = "habitat-testing"
+$TestStartTime = Get-Date
+if ($IsLinux -Or $IsMacOS) {
+    $env:SystemDrive = "/"
+}
+
+function Wait-True([ScriptBlock]$TestScript, [ScriptBlock]$TimeoutScript, [int]$Timeout) {
+    $startTime = Get-Date
+    while (!$TestScript.Invoke()) {
+        $timeTaken = [DateTime]::Now.Subtract($startTime)
+        if ($timeTaken.TotalSeconds -ge $Timeout) {
+            $TimeoutScript.Invoke()
+            break
+        }
+        Start-Sleep -Seconds 1
+    }
+}
+
+function Wait-PathUpdatedAfter($Path, $Time, $Timeout) {
+    $testScript = {
+        if (!(Test-Path -Path $Path)) {
+            return $False
+        }
+        (Get-Item -Path $Path).LastWriteTime -ge $Time
+    }
+    $timeoutScript = { Write-Error "Timed out waiting $Timeout seconds for '$Path' to be updated after '$($Time.ToString("yyyy-MM-ddTHH:MM:ssZ"))'" }
+    Wait-True -TestScript $testScript -TimeoutScript $timeoutScript -Timeout $Timeout
+}
+
+function Test-Port([string]$ComputerName, [int]$Port) {
```

Comment:
we should be able to do away with this abstraction now that we are all core

Reply:
Oddly enough, `Test-Connection` does not seem to work on windows even in powershell core (it always returns false), but `Test-NetConnection` seems to work fine? Any suggestions?

Reply:
weird. I just did some local testing and it does fail with `localhost` but passes with `127.0.0.1`